File: Data Structures & Algorithms/combination-target-sum/submission-2.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        output = []
        stack = [([], 0, 0)]
        while stack:
            subset, cur_sum, idx = stack.pop()
            if cur_sum == target:
                output.append(subset.copy())
                continue
            if idx < len(nums):
                if cur_sum + nums[idx] <= target:
                    stack.append((subset + [nums[idx]], cur_sum + nums[idx], idx))
                stack.append((subset, cur_sum, idx + 1))
        return output

File: Data Structures & Algorithms/remove-element/submission-2.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int offset = 0;
        for (int i=0; i<nums.size(); i++) {
            if (nums[i] == val) {
                offset++;
            }
            else if (offset) {
                nums[i - offset] = nums[i];
            }
        }
        return nums.size() - offset;
    }
};